api/src/test/java/org/openmrs/AllergyTest.java

```diff
@@ -0,0 +1,269 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class AllergyTest {
+
+	@Test(expected = NullPointerException.class)
+	public void hasSameValues_otherAllergyIsNull_shouldThrowNullPointerException() {
+		Allergy allergy = new Allergy();
+		allergy.hasSameValues(null);
+	}
+
+	@Test
+	public void hasSameValues_checkingAgainstSameInstanceOfAllergy_shouldInterpretAsSameValues() {
+		Allergy allergy = allergy();
+		allergy.hasSameValues(allergy());
+	}
+
+	@Test
+	public void hasSameValues_allergiesWithNullIds_shouldBeInterpretedAsSameValues() {
+		assertThat(allergyWithId(null).hasSameValues(allergyWithId(null)), is(true));
+	}
+
+	@Test
+	public void hasSameValues_allergiesWithDifferentIds_shouldBeInterpretAsNotSameValues() {
+		assertThat(allergyWithId(1).hasSameValues(allergyWithId(2)), is(false));
+	}
+
+	@Test
+	public void hasSameValues_allergyWithNullIdAndAllergyWithId_shouldBeInterpretAsNotSameValues() {
+		assertThat(allergyWithId(null).hasSameValues(allergyWithId(1)), is(false));
+	}
+
+	@Test
+	public void hasSameValues_allergyWithIdAndAllergyWithNullId_shouldBeInterpretAsNotSameValues() {
+		assertThat(allergyWithId(1).hasSameValues(allergyWithId(null)), is(false));
+	}
+
+	@Test
+	public void hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() {
+		Allergy allergy = allergyWithPatientId(1);
+		Allergy otherAllergy = allergyWithPatientId(2);
+
+		assertThat(allergy.hasSameValues(otherAllergy), is(false));
```

Comment:
hasSameValues_allergyWithNullIdAndAllergyWithId_shouldBeInterpretAsNotSameValues() and hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() has logically the same code though the name of the functions refer completely opposite to each other. I suggest either to make changes to the 2nd function or have it removed.

Reply:
alryt @haripriya999 , ill look at it

Reply:
no @haripriya999 , they are completey different logically.
with   hasSameValues_allergyWithNullIdAndAllergyWithId_shouldBeInterpretAsNotSameValues() 
the Allery =allergyWithId(*)

yet wtih  hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() 
the Allergy = allergyWithPatientId(*);

allergyWithPatientId(patient id) and  allergyWithId(id) are two different functions

Reply:
Oh sorry, my bad! That part of code seemed a bit unclear to me :)